### PR TITLE
AG-17060 more columnModel tests

### PR DESCRIPTION
**Pull request by SalvatorePreviti**
- add in the column snapshot more states, and update all the snapshot of the tests
- add more api and edge case tests for column model

### CODE REVIEW

**Review by sdwvit — APPROVED** [review]

File: testing/behavioural/src/test-utils/gridColumns/columns-validation-dom/gridColumnsDomValidator.ts [hunk 1]
```diff
@@ -150,15 +150,32 @@ export class GridColumnsDomValidator {
...
+                    colErrors.add(
+                        `Column is sortable but not sorted, aria-sort should be "none" or absent, got "${ariaSort}".`
+                    );
+                }
+            } else if (!isSortable) {
+                // Non-sortable columns should NOT have aria-sort attribute at all
+                // (sort state may exist from colDef but DOM shouldn't expose it)
+                if (ariaSort != null && ariaSort !== 'none') {
+                    colErrors.add(
+                        `Column is not sortable but has aria-sort="${ariaSort}" (should be absent or "none").`
```
[the remark this answers is not in the document]
Comment (SalvatorePreviti) [reply testing/behavioural/src/test-utils/gridColumns/columns-validation-dom/gridColumnsDomValidator.ts]: fixed

File: testing/behavioural/src/columns/column-edge-cases.test.ts [hunk 2]
```diff
@@ -0,0 +1,1527 @@
...
+            const api = gridsManager.createGrid('myGrid', {
+                columnDefs: [{ colId: 'a', width: 1, minWidth: 1 }],
+            });
+
+            expect(api.getColumn('a')!.getActualWidth()).toBe(1);
+            await new GridColumns(api, 'tiny width').checkColumns(false);
+        });
+
+        test('minWidth greater than width clamps up', async () => {
+            const api = gridsManager.createGrid('myGrid', {
```
[the remark this answers is not in the document]
Comment (SalvatorePreviti) [reply testing/behavioural/src/columns/column-edge-cases.test.ts:1132]: fixed

**Review by StephenCooper — APPROVED** [review]
